dcpy/connectors/edm/recipes.py

```diff
@@ -77,210 +82,189 @@ def get_latest_version(name):
+def recipe_from_yaml(path: Path) -> Recipe:
     """Import a recipe file from yaml, and validate schema."""
-    v = cerberus.Validator(RECIPE_FILE_SCHEMA)
     with open(path, "r", encoding="utf-8") as f:
         s = yaml.safe_load(f)
-    if v.validate(s):
-        return s
-    else:
-        raise Exception(f"Invalid recipe file: {path}. {v.errors}")
+    return Recipe(**s)
 
 
-if __name__ == "__main__":
-    flags_parser = argparse.ArgumentParser()
-    flags_parser.add_argument("cmd")
-    cmd = sys.argv[1]
-
-    if cmd == "plan":
-        logger.info("Planning recipe")
-        flags_parser.add_argument(
-            "-f", "--recipe-file", help="Recipe file path", type=Path
-        )
-        flags_parser.add_argument(
-            "-o",
-            "--lock-file",
-            help="Lockfile path",
-            type=Path,
-            required=False,
-        )
-        flags = flags_parser.parse_args()
-
-        recipe = plan_recipe(Path(flags.recipe_file))
-        if flags.lock_file is not None:
-            with open(flags.lock_file, "w", encoding="utf-8") as f:
-                logger.info(
-                    f"Writing recipe lockfile to {str(flags.lock_file.absolute())}"
-                )
-                yaml.dump(recipe, f)
-        else:
-            print(recipe)
-
-    elif cmd == "import":
-        flags_parser.add_argument(
-            "-f", "--recipe-file", help="Recipe file path", type=Path
-        )
-        flags = flags_parser.parse_args()
+app = typer.Typer(add_completion=False)
 
-        with open(flags.recipe_file, "r", encoding="utf-8") as f:
-            recipe = yaml.safe_load(f)
-        logger.info("Importing Recipes")
-        import_recipe_datasets(recipe)
+_typer_recipe_file_opt = typer.Option(
+    None, "--recipe-file", "-f", help="Recipe file path"
+)
 
-    elif cmd == "purge-recipe-cache":
-        logger.info(f"Purging local recipes from {LIBRARY_DEFAULT_PATH}")
-        purge_recipe_cache()
 
-    elif cmd == "write-source-data-versions":
-        flags_parser.add_argument(
-            "-f", "--recipe-file", help="Recipe file path", type=Path
-        )
-        flags = flags_parser.parse_args()
+@app.command()
+def plan(
+    recipe_file: Path = _typer_recipe_file_opt,
+    lock_file: Path = typer.Option(
+        None,
+        "-o",
+        "--lock-file",
+        help="Lockfile path",
+    ),
+):
+    logger.info("Planning recipe")
 
-        recipe = recipe_from_yaml(flags.recipe_file)
-        source_data_versions_path = (
-            flags.recipe_file.parent / "source_data_versions.csv"
+    recipe = plan_recipe(recipe_file)
+    if lock_file is not None:
+        with open(lock_file, "w", encoding="utf-8") as f:
+            logger.info(f"Writing recipe lockfile to {str(lock_file.absolute())}")
+            yaml.dump(recipe.model_dump(), f)
+    else:
+        print(recipe)
+
+
+@app.command("import")
+def import_datasets(recipe_file: Path = _typer_recipe_file_opt):
```

Comment:
Just making sure I understand intended CLI functionality - this function is only supposed to take in a "planned" recipe, right? since it calls `recipe_from_yaml` which just reads a file without looking up any versions for sources, then calls `import_dataset` which requires that versions are not `"latest"`. So then in python or in bash, you should plan, then call this with the name of the "planned" lockfile

Reply:
@fvankrieken Yes, that's the gist. The idea is just that we should resolve versions to actual versions (not latest) before we do any importing. Though honestly, not sure the lockfile is something we want to keep doing.

Reply:
seems nice, like `requirements.in `-> `requirements.txt`

Reply:
Cool. Just might also be good to somehow specify that this "import" command doesn't just take any recipe, but a "planned" recipe

Reply:
@fvankrieken @damonmcc Made it a little more explicit [here](https://github.com/NYCPlanning/data-engineering/pull/232/commits/99484dcd9c0cb928de4f11a37c7772d779c3df57)